lib/reactotron-react-native/src/reactotron-react-native.ts

```diff
@@ -33,13 +33,23 @@ let tempClientId: string | null = null
  *
  * On an Android emulator, if you want to connect any servers of local, you will need run adb reverse on your terminal. This function gets the localhost IP of host machine directly to bypass this.
  */
-const getHost = (defaultHost = "localhost") =>
-  typeof NativeModules?.SourceCode?.getConstants().scriptURL === "string" // type guard in case this ever breaks https://github.com/facebook/react-native/blob/main/packages/react-native/Libraries/NativeModules/specs/NativeSourceCode.js#L15-L21
-    ? NativeModules.SourceCode.scriptURL // Example: 'http://192.168.0.100:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.helloworld'
-        .split("://")[1] // Remove the scheme: '192.168.0.100:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=com.helloworld'
-        .split("/")[0] // Remove the path: '192.168.0.100:8081'
-        .split(":")[0] // Remove the port: '192.168.0.100'
-    : defaultHost
+const getHost = (defaultHost = "localhost") => {
+  try {
+    // RN Reference: https://github.com/facebook/react-native/blob/main/packages/react-native/src/private/specs/modules/NativeSourceCode.js
+    const scriptURL = NativeModules?.SourceCode?.getConstants().scriptURL
+    if (typeof scriptURL !== "string") throw new Error("Invalid non-string URL")
+
+    // Using a capture group to extract the hostname from a URL
+    const host = scriptURL.match(/^(?:https?:\/\/)?([^/:\s]+)(?::\d+)?(?:[/?#]|$)/)?.[1]
```

Comment:
Test cases: https://regex101.com/r/WIHVuQ/2

Reply:
Future 🤔 ...

We could try to use the built in `URL` class if it has the `host` or `hostname` implemented. This would work in current Expo applications. We could then fall back to the regex if that's not available.

Reasons:

* Using what's there so it stays in line with what the application is already using
* More robust support for URLs - The regex above does not support IPv6.
* Hope that URL parsing gets fully implemented so that it's available across the board without polyfill or external packages and then we can remove the regex.

For now, we'll utilize the regex until we can either import the standardized URL parser or it becomes a first class citizen of the platform. To @lindboe point, having one solution means the potential bug reports will be simpler to triage.

Reply:
Could we put these regex in the codebase as unit tests? That way we can add more tests if we have more URL issues in the future.

Reply:
One thing I like to do with regexes is comments that explain their matching section-by-section (here it seems feasible to do per-capture group) just to make sure everybody's on the same page about what's intended

Unit tests aren't a bad idea either 😄